compiler-core/src/language_server/code_action.rs

```diff
@@ -508,3 +515,114 @@ impl<'ast> ast::visit::Visit<'ast> for LabelShorthandSyntax<'_> {
         }
     }
 }
+
+/// Builder for code action to apply the fill in the missing labelled arguments
+/// of the selected function call.
+///
+pub struct FillInMissingLabelledArgs<'a> {
+    module: &'a Module,
+    params: &'a CodeActionParams,
+    line_numbers: LineNumbers,
+    selected_call: Option<(SrcSpan, &'a FieldMap, &'a [TypedCallArg])>,
+}
+
+impl<'a> FillInMissingLabelledArgs<'a> {
+    pub fn new(module: &'a Module, params: &'a CodeActionParams) -> Self {
+        let line_numbers = LineNumbers::new(&module.code);
+        Self {
+            module,
+            params,
+            line_numbers,
+            selected_call: None,
+        }
+    }
+
+    pub fn code_actions(mut self) -> Vec<CodeAction> {
+        self.visit_typed_module(&self.module.ast);
+
+        if let Some((call_location, field_map, args)) = self.selected_call {
+            let mut missing_labels = field_map
+                .fields
+                .iter()
+                .map(|(l, i)| (i, l))
+                .collect::<HashMap<_, _>>();
+
+            for arg in args.iter() {
+                match arg.implicit {
+                    Some(ImplicitCallArgOrigin::Use | ImplicitCallArgOrigin::IncorrectArityUse) => {
+                        _ = missing_labels.remove(&(field_map.arity - 1))
+                    }
+                    Some(ImplicitCallArgOrigin::Pipe) => _ = missing_labels.remove(&0),
+                    // We do not support this action for functions that have
+                    // already been explicitly supplied an argument!
+                    Some(ImplicitCallArgOrigin::PatternFieldSpread) | None => return vec![],
+                }
+            }
+
+            // If we couldn't find any missing label to insert we just return.
+            if missing_labels.is_empty() {
+                return vec![];
+            }
+
+            let add_labels_edit = TextEdit {
+                range: src_span_to_lsp_range(
+                    SrcSpan {
+                        start: call_location.end - 1,
+                        end: call_location.end - 1,
+                    },
+                    &self.line_numbers,
+                ),
+                new_text: missing_labels
+                    .iter()
+                    .sorted_by_key(|(position, _label)| *position)
+                    .map(|(_, label)| format!("{label}: todo"))
+                    .join(", "),
+            };
+
+            let mut action = Vec::with_capacity(1);
+            CodeActionBuilder::new("Fill in missing labelled arguments")
```

Comment:
The name is a bit long and I think it implies it works when there's already some arguments. "Fill labels" or something? Rust calls it "Fill struct fields"

Reply:
Ok! Sounds good to me